server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/cs/factory/DeployedCacheStoreFactory.java

```diff
@@ -35,32 +43,42 @@ public StoreConfiguration processConfiguration(StoreConfiguration storeConfigura
       if(storeConfiguration instanceof DeployedStoreConfiguration) {
          DeployedStoreConfiguration deployedConfiguration = (DeployedStoreConfiguration) storeConfiguration;
          PersistenceConfigurationBuilder replacedBuilder = deployedConfiguration.getPersistenceConfigurationBuilder();
-         DeployedCacheStoreMetadata deployedCacheStoreMetadata = deployedCacheStores.get(deployedConfiguration.getCustomStoreClassName());
 
-         if(deployedCacheStoreMetadata == null) {
-            throw new IllegalStateException("Could not find Deployed Cache metadata for " + deployedConfiguration.getCustomStoreClassName());
-         }
+         try {
+            DeployedCacheStoreMetadata deployedCacheStoreMetadata = getPromise(deployedConfiguration.getCustomStoreClassName()).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
 
-         StoreConfigurationBuilder replacedStoreBuilder = replacedBuilder.addStore(deployedCacheStoreMetadata.getStoreBuilderClass());
-         replacedStoreBuilder.fetchPersistentState(deployedConfiguration.fetchPersistentState());
-         replacedStoreBuilder.ignoreModifications(deployedConfiguration.ignoreModifications());
-         replacedStoreBuilder.preload(deployedConfiguration.preload());
-         replacedStoreBuilder.purgeOnStartup(deployedConfiguration.purgeOnStartup());
-         replacedStoreBuilder.shared(deployedConfiguration.shared());
-         replacedStoreBuilder.withProperties(deployedConfiguration.properties());
-         StoreConfiguration replacedConfiguration = (StoreConfiguration) replacedStoreBuilder.create();
+            StoreConfigurationBuilder replacedStoreBuilder = replacedBuilder.addStore(deployedCacheStoreMetadata.getStoreBuilderClass());
+            replacedStoreBuilder.fetchPersistentState(deployedConfiguration.fetchPersistentState());
+            replacedStoreBuilder.ignoreModifications(deployedConfiguration.ignoreModifications());
+            replacedStoreBuilder.preload(deployedConfiguration.preload());
+            replacedStoreBuilder.purgeOnStartup(deployedConfiguration.purgeOnStartup());
+            replacedStoreBuilder.shared(deployedConfiguration.shared());
+            replacedStoreBuilder.withProperties(deployedConfiguration.properties());
+            StoreConfiguration replacedConfiguration = (StoreConfiguration) replacedStoreBuilder.create();
 
-         return replacedConfiguration;
+            return replacedConfiguration;
+         } catch (Exception e) {
+            InfinispanLogger.ROOT_LOGGER.error("Could not get Custom Cache Store metadata (" + deployedConfiguration.getCustomStoreClassName() + ") within given time.", e);
```

Comment:
Yes I understand that, however what if line 49-58 threw an exception for whatever reason, it would be quite confusing.

To be honest the reason this stuck out to me was because we are catching Exception.  Maybe instead of catching Exception we caught the TimeoutException and ExecutionException separately (the latter shouldn't happen)?  Or do we want to prevent other exceptions as well?